src/extension_utils/utils.ts

```diff
@@ -607,47 +592,38 @@ export const setupEnv = async (
-                                if (
-                                    !(await installDependencies(
-                                        context,
-                                        pythonExecutableName
-                                    ))
-                                ) {
-                                    vscode.window
-                                        .showErrorMessage(
-                                            CONSTANTS.ERROR
-                                                .DEPENDENCY_DOWNLOAD_ERROR,
-                                            DialogResponses.READ_INSTALL_MD
-                                        )
-                                        .then(
-                                            (
-                                                selection:
-                                                    | vscode.MessageItem
-                                                    | undefined
-                                            ) => {
-                                                if (
-                                                    selection ===
-                                                    DialogResponses.READ_INSTALL_MD
-                                                ) {
-                                                    open(
-                                                        CONSTANTS.LINKS.INSTALL
-                                                    );
-                                                }
+                                await installDependenciesWrapper(
+                                    context,
+                                    pythonExecutableName
+                                );
+                            } else {
+                                await vscode.window
+                                    .showInformationMessage(
+                                        CONSTANTS.INFO.ARE_YOU_SURE,
+                                        DialogResponses.INSTALL_NOW,
+                                        DialogResponses.DONT_INSTALL
+                                    )
+                                    .then(
+                                        async (
+                                            installChoice2:
+                                                | vscode.MessageItem
+                                                | undefined
+                                        ) => {
+                                            if (
+                                                installChoice2 ===
+                                                DialogResponses.INSTALL_NOW
+                                            ) {
+                                                await installDependenciesWrapper(
+                                                    context,
+                                                    pythonExecutableName
+                                                );
```

Comment:
If you don't install the dependencies (perhaps you miss the prompts) and try running some micro:bit code, VS Code only shows a generic error:
```
[ERROR] Traceback (most recent call last):
  File "c:\Users\user\.vscode\extensions\vscode-python-devicesimulator\out\process_user_code.py", line 26, in <module>
    from common.telemetry import telemetry_py
  File "c:\Users\user\.vscode\extensions\vscode-python-devicesimulator\out\common\telemetry.py", line 1, in <module>
    from applicationinsights import TelemetryClient
ModuleNotFoundError: No module named 'applicationinsights'
```
It would be much more helpful if we could show the user a more helpful message.

Reply:
True! Perhaps that'd be good for another release, since we're rushing for the release in an hour :'''(